packages/@lwc/synthetic-shadow/src/faux-shadow/events.ts

```diff
@@ -206,17 +206,24 @@ export function addCustomElementEventListener(
     _options?: boolean | AddEventListenerOptions
 ) {
     if (process.env.NODE_ENV !== 'production') {
-        if (!isFunction(listener)) {
+        if (!(isFunction(listener) || isFunction(listener.handleEvent))) {
```

Comment:
```suggestion
        if (!(isFunction(listener) || isFunction(listener?.handleEvent))) {
```

The `listener` could be undefined/null.